.github/workflows/ci.yaml

```diff
@@ -21,6 +21,6 @@ jobs:
         restore-keys: |
           ubuntu-go-
     - name: Test
-      run: go test -mod readonly -race ./...
+      run: go test -race ./...
     - name: Test Verbose
-      run: go test -mod readonly -race -v ./...
+      run: go test -race -v ./...
```

Comment:
I'm confused. usually you run two versions of `go test`, one normal and one with -race. I've never seen anyone run the same tests twice, but one with `-v`.

I'd personally just do `go test ./...` and `go test -race ./...`, is there any reason not to?

As to why run the tests without the race detector first - the race detector tends to slow things down, so if there's an obvious failure, it allows CI to fail much quicker. If the tests pass without the race detector but fail with them, it's also very quick to see that you have a data race, versus a regular bug.

Reply:
FYI my expectation would be that you would have nested tests, e.g. a test which uses testscript to run `go test -v` even if the original call was just `go test`.